plugins/pixart-rf/fu-pxi-firmware.c

```diff
@@ -38,6 +39,9 @@ static gboolean
 fu_pxi_firmware_check_magic(FuFirmware *firmware, GBytes *fw, gsize offset, GError **error)
 {
 	guint64 magic = 0;
+	
+	if (fu_pxi_firmware_get_is_hpac_flag(FU_PXI_FIRMWARE(firmware)))
+		return TRUE;
```

Comment:
is the magic not present in HPAC F/W?  Or is there any other byte to distinguish HPAC F/W?

Ideally what should be happening is that `fwupdtool firmware-parse foo.bin` should be able to match both HPAC and pixart RF F/W.

Reply:
The magic number is not present in HPAC F/W, the file of HPAC F/W is different from other.
The file extension of HPAC F/W is ".fw2", the other F/W is ".bin".

Reply:
Is there something different about HPAC F/W that _can_ be detected?  A different footer?  Or it's really just file name extension?

Reply:
This HPAC F/W is generated by their own HP tool. So we just only can identify HPAC F/W from file extension. When data is transmitted to the device, the device is processing some additional processes, the specific details of which are unknown to us.

Reply:
Can you ask HP for any other details they can share? Perhaps their tool leaves some kind of signature and it's just not known right now.

Reply:
May I ask is there have any affect if we can't parse the HPAC F/W? Because we can add the device with PID/VID and private flag in .quirk to identify it and do the firmware update. I will check with them.

Reply:
@hughsie @superm1  I have added the header judgement for HPAC in commit a3f8969ce1bb3ddf34e463aa232dc6861f6f6d3e , we add the footer in HPAC F/W file.

I still keep the private flag, this can be used for firmware double checking.